### ISSUE TITLE
[GAMEPLAY] Add weapon auto-switch on ammo running out

### ISSUE THREAD

**Issue by ObaniGemini** [Siile/Ninslash#36]
If weapon runs out of ammo, and that you shoot one time while there's no ammo, it's a common usage to auto-switch to the previous/next weapon you have (in quake games mostly).
_Teeworlds_ doesn't have many weapons, and it's gameplay is less messy (one bullet for one action) than ninslash's one. That's why it doesn't need an auto-switch option (imo).

But in _Ninslash_, you run out of ammo quickly without noticing it but you have lots of weapons to continue a fight, and there are so much visual effects that it is way harder to distinct it when you run out of ammo. That's why I think it's a necessity to have an auto-switch option (imo).

P.S : I think auto-switch should be an option set to "true" by default, probably located in "general settings".

**Comment by Siile** [comment]
This has been proposed before, and it's definitely worth trying out.

**Comment by pelya** [comment]
Android version has this feature, I'll submit a patch when I'll have some free time.

**Comment by ObaniGemini** [comment]
@pelya : is it actually an option accessible into GUI ?

**Comment by pelya** [comment]
Yes, in Android version, Settings - General - Switch weapon when out of
ammo.
